.circleci/config.yml

```diff
@@ -226,16 +212,35 @@ jobs:
       - *make
       - *test
 
+  "validate-i386-linux":
+    resource_class: xlarge
+    docker:
+      - image: mrkkrp/ghcci-i386-linux:0.0.1
+    environment:
+      <<: *buildenv
+    steps:
+      - checkout
+      - *prepare
+      - *submodules
+      - *boot
+      - *configure_unix_32
+      - *make
+      - *test
+      - *bindist
+      - *collectartifacts
+      - *storeartifacts
+
 workflows:
   version: 2
   validate:
     jobs:
-    - validate-x86_64-linux
+    # - validate-x86_64-linux
     # FreeBSD disabled: https://github.com/haskell/unix/issues/102
     # - validate-x86_64-freebsd
-    - validate-x86_64-darwin
-    - validate-x86_64-linux-llvm
-    - validate-hadrian-x86_64-linux
+    # - validate-x86_64-darwin
+    # - validate-x86_64-linux-llvm
+    - validate-i386-linux
+    # - validate-hadrian-x86_64-linux
```

Comment:
I would be happy to merge this and fix up the issues in a later PR if this were removed.